lib/_http_server.js

```diff
@@ -521,9 +521,12 @@ function socketOnError(e) {
     if (this.writable) {
       const response = e.code === 'HPE_HEADER_OVERFLOW' ?
         requestHeaderFieldsTooLargeResponse : badRequestResponse;
-      this.write(response);
+      this.write(response, () => {
+        this.destroy(e);
+      });
```

Comment:
Unfortunately I don't think there is a guarantee that this callback is actually called.